### DISCUSSION TITLE
How to continue the script even after 1 host fails from the list

### DISCUSSION THREAD

**Discussion by sagarCCIE** [ktbyers/netmiko#2210] [General]
Hello Kirk, Team
I have list of 10 devices and if 1 from center fails, Netmiko scripts stop.
Any workaround to skip the failed one and keep continuing till the end.
Please assist

**Comment by ktbyers** [comment]
@sagarCCIE Usually, you just add a try/except statement and keep going if one device fails.

You could then add some Python logging indicating information about the failing device.


Regards,

Kirk

**Reply by sagarCCIE** [reply]
Thank you.
Can you point me to a readymade script which you have in your mind for me to refer.

**Comment by Microbuse** [comment]
Hi. 
I changed in the base_connection.py line 957
from raise NetmikoTimeoutException(msg) to print(NetmikoTimeoutException(msg)) and added only raise om next line.
The it only logs the error connection and move to next device. in line

**Reply by Microbuse** [reply]
![base_change](https://user-images.githubusercontent.com/81070908/129874743-2a4880d3-5355-410b-aa1c-833e0fc54e45.JPG)

**Comment by ktbyers** [comment]
In Netmiko 4.x (currently develop branch), I would do the following:

```python
import logging
from netmiko import ConnLogOnly
from devices import cisco3, cisco4, arista1, arista2

log_level = logging.INFO
log_file = "my_output.log"

for device in (cisco3, cisco4, arista1, arista2):
    net_connect = ConnLogOnly(log_file=log_file, log_level=log_level, **device)
    if net_connect:
        print(net_connect.find_prompt())
```

`ConnLogOnly` will cause a device to not create exceptions on connection failures--instead these will get logged to the logging file that you specify.
